AdobeStockStub/Model/FileGenerator.php

```diff
@@ -0,0 +1,243 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockStub\Model;
+
+/**
+ * Generate a stub Adobe Stock API Asset file object based on the search query.
+ */
+class FileGenerator
+{
+    /**
+     * Generate an array of Adobe Stock stub files array based on the query.
+     *
+     * @param array $query
+     *
+     * @return array
+     */
+    public function generate(array $query): array
```

Comment:
I don't think the generate method need `$query`. Generator should return the default data unless overrided from the resolver.
I propose to introduce an `$itemOverride` array containing the field values that should be merged for each item and `$dataOverride` array containing the array that should be merged to the final result
```suggestion
    public function generate(array $dataOverride, array $itemOverride, int $recordsCount): array
```

I don't think there will be need to process each `search_parameters` item